src/pages/HomePage/index.tsx

```diff
@@ -58,14 +78,38 @@ export default function HomePage() {
     setCurrentPage(currentPage - 1);
   };
 
+  const handleClickSearch = async () => {
+    if (isAllowedToSearch && searchText) {
+      const result = await searchTransactionQuery({
+        variables: { transaction: searchText, address: searchText },
+      });
+
+      if (result.data?.transaction?.id) {
+        window.location.assign(`/transaction/${searchText}`);
```

Comment:
It's possible to use the router system here?

Reply:
I'll check

Reply:
done.